src/components/MultiHopTrade/hooks/useSupportedAssets.tsx

```diff
@@ -0,0 +1,52 @@
+import { useMemo } from 'react'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { LifiSwapper } from 'lib/swapper/swappers/LifiSwapper/LifiSwapper'
+import { selectAssetsSortedByMarketCapFiatBalanceAndName } from 'state/slices/common-selectors'
+import { selectAssetIds, selectSellAsset } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+export const useSupportedAssets = () => {
+  const sellAsset = useAppSelector(selectSellAsset)
+  const assetIds = useAppSelector(selectAssetIds)
+  const sortedAssets = useAppSelector(selectAssetsSortedByMarketCapFiatBalanceAndName)
+  const isLifiEnabled = useFeatureFlag('LifiSwap')
+
+  const enabledSwappers = useMemo(() => {
+    const result = []
+    if (isLifiEnabled) result.push(LifiSwapper)
+    // TODO(woodenfurniture): add more swappers here
+    return result
+  }, [isLifiEnabled])
+
+  const supportedSellAssets = useMemo(() => {
+    let supportAssetIds = assetIds
+
+    for (const { filterAssetIdsBySellable } of enabledSwappers) {
+      // TODO(woodenfurniture): filter Asset[] instead so filtering with a set below isnt required
+      supportAssetIds = filterAssetIdsBySellable(supportAssetIds)
+    }
```

Comment:
or a non-mutatey version:

```suggestion
    const supportAssetIds = enabledSwappers.reduce((prevAssetIds, { filterAssetIdsBySellable }) => {
        // TODO(woodenfurniture): filter Asset[] instead so filtering with a set below isnt required
+      return filterAssetIdsBySellable(prevAssetIds)
+    }, assetIds)
```

Reply:
I love this. My initial implementation was for performance reasons because we were originally passing in the full set of asset IDs for every iteration, and my change was to pass in the reduced set each iteration.

Love your version, much nicer